cmd/driver-registrar/main.go

```diff
@@ -110,6 +156,45 @@ func main() {
 	}
 	glog.V(2).Infof("CSI driver node ID: %q", csiDriverNodeId)
 
+	// When kubeletRegistrationPath is specified then driver-registrar ONLY acts
+	// as gRPC server which replies to registration requests initiated by kubelet's
+	// pluginswatcher infrastructure. Node labeling is done by kubelet's csi code.
+	if *kubeletRegistrationPath != "" {
+		registrar := newRegistrationServer()
+		registrar.driverName = csiDriverName
+		registrar.endpoint = *kubeletRegistrationPath
+		registrar.version = supportedVersions
+		socketPath := fmt.Sprintf("/registration/%s-reg.sock", csiDriverName)
+		// Remove any socket, stale or not, but fall through for other files
+		fi, err := os.Stat(socketPath)
+		if err == nil && (fi.Mode()&os.ModeSocket) != 0 {
+			os.Remove(socketPath)
```

Comment:
What happens if `os.Remove` fails? We should explicitly handle the error.

Reply:
done